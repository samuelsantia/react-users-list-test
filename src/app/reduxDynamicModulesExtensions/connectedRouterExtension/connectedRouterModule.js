import { connectRouter } from 'connected-react-router';

/**
 * redux-dynamic-moduler router module
 * loads the router reducer
 * @param {History} history router history
 * @return {Object} router module config
 */
export const getRouterModule = (history) => ({
  id: 'router',
  reducerMap: {
    router: connectRouter(history),
  },
});
