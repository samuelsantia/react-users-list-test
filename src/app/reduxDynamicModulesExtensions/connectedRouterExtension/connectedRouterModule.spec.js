import { connectRouter } from 'connected-react-router';
import { getRouterModule } from './connectedRouterModule';

jest.mock('connected-react-router');

const fakeRouterReducer = (value) => value;
const fakeHistory = {};

beforeEach(() => {
  connectRouter.mockReturnValue(fakeRouterReducer);
});

afterEach(() => {
  jest.resetAllMocks();
});

describe('getRouterModule', () => {
  it('should return object id `router`', () => {
    const actual = getRouterModule(fakeHistory).id;
    const expected = 'router';

    expect(actual).toBe(expected);
  });

  it('should has reducerMap with connected router reducer', () => {
    const actual = getRouterModule(fakeHistory).reducerMap;
    const expected = { router: fakeRouterReducer };

    expect(connectRouter).toHaveBeenCalledTimes(1);
    expect(connectRouter).toHaveBeenCalledWith(fakeHistory);
    expect(actual).toEqual(expected);
  });

  it('should match snapshot', () => {
    expect(getRouterModule(fakeHistory)).toMatchSnapshot();
  });
});
