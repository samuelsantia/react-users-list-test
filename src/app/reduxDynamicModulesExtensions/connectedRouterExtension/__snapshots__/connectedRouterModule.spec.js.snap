// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRouterModule should match snapshot 1`] = `
Object {
  "id": "router",
  "reducerMap": Object {
    "router": [Function],
  },
}
`;
